apps/stable_diffusion/scripts/txt2img.py

```diff
@@ -160,19 +167,32 @@ def txt2img_inf(
         seeds.append(img_seed)
         txt2img_obj.log += "\n"
 
-    total_time = time.time() - start_time
-    text_output = f"prompt={args.prompts}"
-    text_output += f"\nnegative prompt={args.negative_prompts}"
-    text_output += f"\nmodel_id={args.hf_model_id}, ckpt_loc={args.ckpt_loc}"
-    text_output += f"\nscheduler={args.scheduler}, device={device}"
-    text_output += (
-        f"\nsteps={steps}, guidance_scale={guidance_scale}, seed={seeds}"
-    )
-    text_output += f"\nsize={height}x{width}, batch_count={batch_count}, batch_size={batch_size}, max_length={args.max_length}"
-    text_output += txt2img_obj.log
-    text_output += f"\nTotal image generation time: {total_time:.4f}sec"
+        total_time = time.time() - start_time
+        text_output = f"prompt={args.prompts}"
+        text_output += f"\nnegative prompt={args.negative_prompts}"
+        text_output += (
+            f"\nmodel_id={args.hf_model_id}, ckpt_loc={args.ckpt_loc}"
+        )
+        text_output += f"\nscheduler={args.scheduler}, device={device}"
+        text_output += (
+            f"\nsteps={steps}, guidance_scale={guidance_scale}, seed={seeds}"
+        )
+        text_output += f"\nsize={height}x{width}, batch_count={batch_count}, batch_size={batch_size}, max_length={args.max_length}"
+        text_output += (
+            f"\n\nTotal image(s) generation time: {total_time:.4f}sec"
+        )
+        print(txt2img_obj.log)
+        txt2img_obj.log = ""
+        if i != batch_count - 1:
+            stable_diffusion_button_label = (
+                f"Processing batch {i+2} / {batch_count}"
+            )
```

Comment:
It would be better to disable the generate button instead of changing the labels.

Reply:
Labels are now kept unchanged.
A clear queue button was also added to cancel batches.
I gave up on tempering with buttons status, I spent more than 10 hours on it, it's a nightmare:
First it's not just buttons, prompt and neg prompt are also trigerring jobs, and what about fields from the other tabs ?
More over if we lock them, we will also need to unlock them.

I checked Auto1111, this is the direction they took. 
They added a state manager and update states on every possible actions, more over this is not working very well.
https://www.reddit.com/r/StableDiffusion/comments/y0mhkt/is_there_any_fix_for_the_gradio_bug_that_prevents/